benchmarks/transformer/score_mod.py

```diff
@@ -257,18 +260,28 @@ def relative_bias(score, b, h, m, n):
     def head_bias(score, b, h, m, n):
         return score + 2 * h
 
-    return [noop, causal_mask, relative_bias, head_bias]
-
-
-def generate_experiment_configs(calculate_bwd: bool) -> List[ExperimentConfig]:
-    batch_sizes = [2, 8, 16]
-    num_heads = [16]
-    q_kv_seq_lens = [(512, 512), (1024, 1024), (4096, 4096)]
-    head_dims = [64, 128]
+    function_dict = {
+        "noop": noop,
+        "causal": causal_mask,
+        "rel": relative_bias,
+        "head_bias": head_bias,
+    }
+    return [function_dict[name] for name in score_mods]
+
+
+def generate_experiment_configs(
+    calculate_bwd: bool,
+    batch_sizes: List[int],
+    num_heads: List[int],
+    seq_lens: List[int],
+    head_dims: List[int],
+    score_mods: List[str],
+) -> List[ExperimentConfig]:
+    q_kv_seq_lens = [(i, i) for i in seq_lens]  # only testing q_len == kv_len
     dtypes = [
-        torch.bfloat16,
+        torch.float16,
```

Comment:
Is bf16 more common in real use case?

Reply:
bf16 is probably more common but I was thinking that perhaps we should just benchmark with fp16, since it has better perf on A100 haha.